examples/fit_ODNP_data.py

```diff
@@ -0,0 +1,174 @@
+# }}}
+with figlist_var() as fl:
+    # {{{Plot integrals as a function of power
+    fl.next("Integrals vs power")
+    fl.plot(
+        integral_vs_p["power", :flip_idx],
+        "o",
+        label="Progressive Saturation Data",
+        capsize=6,
+        alpha=0.5,
+    )
+    fl.plot(
+        integral_vs_p["power", flip_idx:],
+        "rx",
+        label="Returning Power Check",
+        capsize=6,
+        alpha=0.5,
+    )
+    # }}}
+    # {{{ plot R1p data
+    fl.next(r"$R_{1}(p)$")
+    fl.plot(R1p, "o", label="Experimental Data")
+    # }}}
+    # {{{ fit kᵨ⁻¹ with two degrees of freedom (to a straight line) and then
+    # apply to fit R1p
+    T10_p = loadmat(
+        search_filename(
+            T100_info["filename"], exp_type=T100_info["data_dir"], unique=True
+        )
+    )["a"][0, :]
+    R10_p = 1 / (R1p.fromaxis("power").eval_poly(T10_p, "power"))
+    powers_fine = nddata(
+        r_[0 : R1p.getaxis("power")[-1] : 300j], "p"
+    )  
+    krho_inv = integral_vs_p.get_prop("acq_params")["concentration"] / (R1p - R10_p)
+    krho_inv_coeff = krho_inv.polyfit("power", order=1)
+    M0, A, phalf, p = symbols("M0 A phalf power", real=True)
+    R1p_expression = (T10_p[0] + T10_p[1] * p) ** -1 + (
+        integral_vs_p.get_prop("acq_params")["concentration"]
+        / (krho_inv_coeff[0] + krho_inv_coeff[1] * p)
+    )
+    R1p_fit = lambdify(p, R1p_expression)
+    fl.plot(
+        R1p_fit(powers_fine),
+        color="k",
+        label="Fit",
+        alpha=0.5,
+    )
+    plt.ylabel(r"$R_{1} / s^{-1}$")
+    plt.xlabel("Power / W")
+    # }}}
+    # {{{ Fit E(p)
+    fl.next("Integrals vs power")
+    sp = p / (p + phalf)
+    integral_vs_p_fit = lmfitdata(integral_vs_p["power", :flip_idx])
+    # Symbolic expression for integrals as a function of power that is used 
+    # in the symbolic function for the fitting of the integrals as a function of power
+    integral_vs_p_fit.functional_form = M0 - ((M0 * A * sp) / R1p_expression)
+    # generate a guess for the A parameter of the fit based on the normalized
+    # enhancement weighted by the relaxation rate. The bounds for the fit are
+    # then set to center around this value.
+    # Since
+    # E(pₘₐₓ) = 1 - A s(pₘₐₓ)/R₁(pₘₐₓ)
+    # and the max s(p) is about 1,
+    # 1-E(pₘₐₓ)R₁(pₘₐₓ) ≈ A
+    # because R₁(0) is typically larger and also easier to access than
+    # R₁(pₘₐₓ), we just use it instead.
+    A_guess = (
+        1
+        - (
+            R1p["power", 0].real.item()
+            * (
+                integral_vs_p["power", flip_idx].real.item()
+                /integral_vs_p["power",0].real.item()
+            )
+        ).real
+    )
+    integral_vs_p_fit.set_guess(
+        M0=dict(value=integral_vs_p["power", 0].real.item(), min=1e4, max=11e4),
+        A=dict(value=A_guess, min=0.2 * A_guess, max=3 * A_guess),
+        phalf=dict(value=0.2, min=0.05, max=1.0),
+    )
+    integral_vs_p_fit.settoguess()
```

Comment:
Set to guess is not required if you are about to call fit. You only need it if you are about to call eval so that you can plug your guess.

Reply:
In principle, I think that we could actually edit the class so that we don't need `set_to_guess` at all… as long as it knew whether we called `set_to_guess` or `fit` last, it should be able to figure out what we want when we call eval... put in your own words and create an issue on pySpecData.